includes/types/CMB2_Type_Associated_Objects.php

```diff
@@ -451,42 +323,44 @@ public function get_object( $id ) {
 	 * @return array            Array of attached object ids.
 	 */
 	public function get_all_objects( $attached = array() ) {
-		$args = $this->query_args;
-		$objects = $this->get_objects( $args );
-
-		$attached_objects = array();
-		foreach ( $objects as $object ) {
-			$attached_objects[ $this->get_id( $object ) ] = $object;
-		}
+		$query = $this->get_query( $this->query_object_type );
+		$objects = $query->execute_query();
 
 		if ( ! empty( $attached ) ) {
-			$is_users = 'user' === $this->field->options( 'query_object_type' );
-			$args[ $is_users ? 'include' : 'post__in' ] = $attached;
-			$args[ $is_users ? 'number' : 'posts_per_page' ] = count( $attached );
+			$query->set_include( $attached );
+			$query->set_number( count( $attached ) );
 
-			$new = $this->get_objects( $args );
+			$new = $query->execute_query();
 
 			foreach ( $new as $object ) {
-				if ( ! isset( $attached_objects[ $this->get_id( $object ) ] ) ) {
-					$attached_objects[ $this->get_id( $object ) ] = $object;
+				if ( ! isset( $objects[ $query->get_id( $object ) ] ) ) {
+					$objects[ $query->get_id( $object ) ] = $object;
 				}
 			}
 		}
 
-		return $attached_objects;
+		return $objects;
 	}
 
 	/**
-	 * Peforms a get_posts or get_users query.
+	 * @param string $query_object_type
 	 *
-	 * @since  1.2.4
-	 *
-	 * @param  array  $args Array of query args.
-	 *
-	 * @return array        Array of results.
+	 * @return CMB2_Type_Associated_Objects_Query
 	 */
-	public function get_objects( $args ) {
-		return call_user_func( 'user' === $this->field->options( 'query_object_type' ) ? 'get_users' : 'get_posts', $args );
+	public function get_query( $query_object_type = 'post' ) {
+		switch ( $query_object_type ) {
+			case 'user';
+				$query = new CMB2_Type_Associated_Objects_Query_Users( $this->query_args );
+				break;
+			case 'term':
+				$query = new CMB2_Type_Associated_Objects_Query_Terms( $this->query_args );
+				break;
+			case 'post':
+			default:
+				$query = new CMB2_Type_Associated_Objects_Query_Posts( $this->query_args );
```

Comment:
This class name is clean, but would maybe prefer a pattern that was a bit more concise. e.g. `CMB2_Type_Associated_Posts_Query`, leaving the base as `CMB2_Type_Associated_Objects_Query`.